draft-bash-rfc7958bis.mkd

```diff
@@ -162,13 +158,18 @@ This is in the seventh paragraph of {{xml_semantics}}.
 
 - The reference to the DNSSEC Practice Statement {{DPS}} was updated.
 
+- Say explicitly that the XML documents might have XML comments in them.
+
 # IANA DNSSEC Root Zone Trust Anchor Formats and Semantics {#ta_formats}
 
 IANA publishes trust anchors for the root zone as an XML document that contains the hashes of the DNSKEY records.
 
 This format and the semantics associated are described in
 the rest of this section.
 
+Note that the XML document can have XML comments.
+For example, IANA might use these comments to add pointers to important information on the IANA web site.
```

Comment:
Should we declare here that the XML comments will not be used to extend the protocol, i.e. the XML comments are purely human-readable commentary?

On the topic of extension, do we have an plan there? The way the document is written, an implementer may decide that the XML should strictly obey the schema. To allow extensibility, we may want to say that unknown tags and elements should be ignored.

I think there is an argument that implementations of this are security infrastructure and should therefore have continuous updates (hence backwards-incompatible extensions are OK). However, there will still be an interim/rollout period, and implementations may be embedded in hardware (e.g. an HSM) that is still useful even when not receiving regular updates.

Reply:
I like the idea of saying explicitly that comments are only intended to be informal guidance to human operators, have no regular format, might appear or disappear at any time, and must not (MUST NOT!) be relied upon for automated trust anchor handling.